projects/eventonica/eventonica-part6-postgres.md

```diff
@@ -1,9 +1,51 @@
 # Eventonica - Part 6 - Postgres Database
 
 ### Overview
-
+Previously, your data was stored in memory in Express, so your data would disappear if the application restarted. For production applications, data must be stored in a more permanent way. In this lesson, you will move your EventRecommender data to a Postgres database and connect to that database in your Express APIs.
 
 ### Instructions
 
+1. Ensure that you have Postgres installed on your machine - see instructions [here](../../databases/installing-postgresql.md).
+
+1. Create a new database named `eventonica`.
+
+1. In your `eventonica` database, create a table named `users` that contains the same fields as your `User` class in `eventRecommender.js`. Make the `id` column a [primary key](https://www.postgresql.org/docs/12/ddl-constraints.html#DDL-CONSTRAINTS-PRIMARY-KEYS) and use the datatype [serial](https://www.postgresql.org/docs/12/datatype-numeric.html#DATATYPE-SERIAL) to create an auto-incrementing integer id. Try running the following SQL insert multiple times to see how the `serial` type works: `INSERT INTO users (name) values ('jane');`. Your table should have automatically filled the `id` field for you!
+
+1. Create a table named `events` that contains the same fields as your `Event` class in `eventRecommender.js`. Create the `id` column like you did for the `users` table.
+
+1. Install [pg-promise](https://expressjs.com/en/guide/database-integration.html#postgresql) in your project folder - this module connects your Express application to a Postgres database.
+
+1. Copy the setup instructions for `pg-promise` in your `index.js` file. Your connection string is probably something like `postgres://{YOURNAME}@localhost:5432/eventonica`. If you used a password to protect your database, use `postgres://{YOURNAME}:{PASSWORD}@localhost:5432/eventonica`.
+
+1. Update your APIs to use SQL commands instead of the EventRecommender class. You can find example queries [here](https://github.com/vitaly-t/pg-promise/wiki/Learn-by-Example). Note that `pg-promise` requires you to specify how many rows, if any, a query should return. For example, `db.any` indicates that the query can return any number of rows, `db.one` indicates that the query should return a single row, and `db.none` indicates that the query must return nothing.
+
+    Ex: Adding a user
+
+    ```
+    # using db.one because the INSERT will return a single new user
+    db.one('INSERT INTO users (name) values ($1) RETURNING id, name', [req.body.name])
+    .then(data => {
+        # on success, return the user data
+        res.send(data)
+    })
+    .catch(function(error) {
+        # on failure, send a generic 500 status
+        res.sendStatus(500)
+    });
+    ```
+
+1. Remove the EventRecommender import and instance from your `index.js` file.
+
+1. Test that your new APIs work using Postman and your webpage. Try restarting the Express application - your data should still be there!
```

Comment:
Is it worth them checking using `psql` that the data is safely in postgres?

Reply:
I don't think it's worth checking but I'll add a sentence to reify that the database is independent of the Express app.